feat(curriculum): Add fCC Authors Page Project to JS Cert (#49655)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/63ff7c3430a5d21ddc1466e6.md

---
id: 63ff7c3430a5d21ddc1466e6
title: Step 9
challengeType: 0
dashedName: step-9
---

# --description--

> Loop through the `authors` with a `forEach` inside your `displayAuthors` function. Don't do any other thing yet. Just attach `forEach()` to `authors`.

Reviewer: ```suggestion
Inside your `displayAuthors` function, chain `forEach()` to `authors`.
```

---
id: 63ff7c3430a5d21ddc1466e6
title: Step 8
challengeType: 0
dashedName: step-8
---

# --description--

The data containing the authors is an array, so you have to loop through it with an array method before you'll be able to populate it on the UI.

The best array method for this purpose is the `forEach()` method. That's because it's ideal when you want to execute a function for each element in an array.

The `forEach()` method takes a callback function that holds arguments you can use within the callback function. This callback can be used for each item such as the array item, the index of the item, and the array itself.

Loop through the `authors` with a `forEach` inside your `displayAuthors` function.

To loop through anything, it normally looks like this:

```js
parameter.arrayMethod(

)

> ```

Reviewer 2: While this explanation is good, I don't think it needs to be as detailed because by this point in the curriculum, campers should have a good amount of experience working with the `forEach` method in earlier projects. 

I think we should just keep this line here

Loop through the `authors` with a `forEach` inside your `displayAuthors` function.

Author: Okay. I will keep only that line Jessica suggested I keep.
